test/functional/kernel/basic-single-loop/CMakeLists.txt

```diff
@@ -25,6 +25,17 @@ foreach( BACKEND ${KERNEL_BACKENDS} )
     target_include_directories(test-kernel-basic-single-loop-${TESTTYPE}-${BACKEND}.exe
                                PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/tests)
   endforeach()
+
```

Comment:
We could consolidate this to cover all tests in one place by adding another nested loop and another variable, right?

set(RESOURCE_USES basic resource-basic)
....

then add innermost loop:

foreach( RESOURCE_USE ${RESOURCE_USES} )
   configure_file( .... test-kernel-${RESOURCE_USE}-single-loop....

Reply:
Yeah, I went back and forth on the current implementation and what you suggested, I didn't really like either and landed on the more verbose of the two. I can go ahead and use this suggested method if that is preferred

Reply:
Sounds like you thought about it more than I have. Keep what you have.